python-sdk/src/astro/files/base.py

```diff
@@ -79,38 +77,13 @@ def create_from_dataframe(self, df: pd.DataFrame) -> None:
         ) as stream:
             self.type.create_from_dataframe(stream=stream, df=df)
 
-    def _convert_remote_file_to_byte_stream(self) -> io.IOBase:
-        """
-        Read file from all supported location and convert them into a buffer that can be streamed into other data
-        structures.
-
-        Due to noted issues with using smart_open with pandas (like
-        https://github.com/RaRe-Technologies/smart_open/issues/524), we create a BytesIO or StringIO buffer
-        before exporting to a dataframe. We've found a sizable speed improvement with this optimizat
-
-        Returns: an io object that can be streamed into a dataframe (or other object)
-
-        """
+    def export_to_dataframe(self, **kwargs) -> pd.DataFrame:
+        """Read file from all supported location and convert them into dataframes."""
         mode = "rb" if self.is_binary() else "r"
-        remote_obj_buffer = io.BytesIO() if self.is_binary() else io.StringIO()
         with smart_open.open(
             self.path, mode=mode, transport_params=self.location.transport_params
         ) as stream:
-            remote_obj_buffer.write(stream.read())
```

Comment:
@pankajkoti when we had the code this way before we had serious speed issues due to known issues with smart_open and pandas. Are you not seeing those speed issues anymore? What has changed?

Reply:
I am not seeing speed issues at least with `csv` files. Have shared the benchmark numbers with this approach in Slack thread. But I'm seeing the process is getting killed otherwise when we're loading large files e.g. 5GB 

cc: @utkarsharma2 @kaxil

Reply:
@dimberman here are the benchmarking results post change - 
Running the analysis on postgres_stats.ndjson...

### Database: postgres

| database   | dataset    | total_time   | memory_rss   | cpu_time_user   | cpu_time_system   |
|:-----------|:-----------|:-------------|:-------------|:----------------|:------------------|
| postgres   | hundred_kb | 8.94s        | 67.7MB       | 594.18ms        | 167.63ms          |
| postgres   | hundred_mb | 29.8s        | 920.42MB     | 4.55s           | 717.26ms          |
| postgres   | one_gb     | 3.69min      | 654.05MB     | 41.27s          | 8.8s              |
| postgres   | one_mb     | 9.72s        | 83.71MB      | 613.84ms        | 168.45ms          |
| postgres   | ten_kb     | 7.38s        | 67.19MB      | 529.94ms        | 154.06ms          |
| postgres   | ten_mb     | 12.1s        | 176.21MB     | 870.54ms        | 195.1ms           |
| postgres   | five_gb   | 18.11min     | 92.29MB      | 3.49min         | 58.18s            |
| postgres   | ten_gb    | 42.29min     | 83.36MB      | 7.19min         | 2.11min           |

Reply:
Also, according to this comment, we only have this issue with smart_open and parquet file - https://github.com/RaRe-Technologies/smart_open/issues/524#issuecomment-663036104

As part of this PR we have moved this logic to work on Parquet files.

Reply:
@utkarsharma2 those look like the results _before_ the optimization. Here are there results after the optimization
<img width="906" alt="Screen Shot 2022-09-08 at 2 08 47 PM" src="https://user-images.githubusercontent.com/2644098/189226342-63deb470-73c5-4946-904f-7034dff00f32.png">

Reply:
@dimberman So they were done for S3 to Postgres running on local actually. But I do get the concern I'll have the numbers for GCS to Postgres on Cloud.

Reply:
@dimberman so if the number matches for the GCS to Postgres on Cloud, do we have you blessing on this one :)

Reply:
These are the comparative findings for postgres load from GCS on GCP Cloud VM:

On main branch:
| database   | dataset    | total_time   | memory_rss   | cpu_time_user   | cpu_time_system   |
|:-----------|:-----------|:-------------|:-------------|:----------------|:------------------|
| postgres   | ten_kb     | 696.19ms     | 32.93MB      | 700.0ms         | 50.0ms            |
| postgres   | hundred_kb | 725.48ms     | 33.71MB      | 720.0ms         | 30.0ms            |
| postgres   | ten_mb     | 1.92s        | 48.55MB      | 1.32s           | 150.0ms           |
| postgres   | one_gb     | 1.64min      | 226.88MB     | 59.77s          | 9.7s              |

With this branch (current PR):
| database   | dataset    | total_time   | memory_rss   | cpu_time_user   | cpu_time_system   |
|:-----------|:-----------|:-------------|:-------------|:----------------|:------------------|
| postgres   | ten_kb     | 783.65ms     | 34.89MB      | 730.0ms         | 40.0ms            |
| postgres   | hundred_kb | 808.62ms     | 35.77MB      | 740.0ms         | 20.0ms            |
| postgres   | ten_mb     | 6.01s        | 52.29MB      | 1.33s           | 130.0ms           |
| postgres   | one_gb     | 4.62min      | 601.58MB     | 1.05min         | 5.63s             |

Reply:
Sync-up with Daniel on that but priority-wise I would like to have great performance for Bigquery, Snowflake and Redshift (all the cloud data warehouses)

Reply:
With the code on `main` branch, what I observe is loading a `SINGLE 5GB` file is failing on the GCP VM instance for all databases `bigquery`, `snowflake`, `postgres` & `redshift`.

Whereas, with this change, although slower the load is working fine.

Reply:
Also, @dimberman , do you remember what was the setup and dataset used for the current benchmarking stats.
I am seeing a significant difference in the stats on main branch than what we have for the real 5GB dataset we have in `gs://astro-sdk/benchmark/pypi/*`. Ran this on a GCP instance with running a local postgres on it.

On main branch:
| database   | dataset   | total_time   | memory_rss   | cpu_time_user   | cpu_time_system   | memory_pss   | memory_shared   |
|:-----------|:----------|:-------------|:-------------|:----------------|:------------------|:-------------|:----------------|
| postgres   | five_gb   | 27.18min     | 501.94MB     | 25.75min        | 28.11s            | 501.64MB     | 7.96MB          |


On current PR branch:
| database   | dataset   | total_time   | memory_rss   | cpu_time_user   | cpu_time_system   | memory_pss   | memory_shared   |
|:-----------|:----------|:-------------|:-------------|:----------------|:------------------|:-------------|:----------------|
| postgres   | five_gb   | 40.71min     | 539.33MB     | 26.15min        | 22.32s            | 539.26MB     | 7.99MB          |